### PR TITLE
Added a backend flask API

### PR DESCRIPTION
**Pull request by sachin-sankar**
As the project is mainly going to be a web application i decided to make an API using flask that will serve a route `/api/problem`.

This route generates a problem set using the newly added `generate_problem()` function in `conversions.py` and send a JSON response containing all data needed for the frontend as of now as seen in the prototype UI by swas. Here is a list of all data included in the response.

- `start_substrate` : Start substrate of the problem
- `end_substrate` : End substrate of the problem
- `start_substrate_image_url` : Substrate structure image url
- `end_substrate_image_url` : Substrate structure image url
- `solution` : Problem solution
- `solution_set` : 3 Random reagent containing the solution reagent also

This JSON then can be plugged into the frontend using javascript. The image url are as it is meaning they take a very low time loading and are from the OPSIN database.

### CODE REVIEW

**Comment by CodeWithSwastik** [comment]
I like this a lot but I'm planning to implement this in FastAPI and not Flask.

File: conversions.py [hunk 1]
```diff
@@ -45,12 +45,24 @@ def generate_conversion_problem(start: Optional[str] = None, length: int = 3):
         end = random.choice(list(reactions[prev].values()))
         
         if c == 10000:
-            break
+        	break
```
Comment (CodeWithSwastik) [inline conversions.py]: Why?

File: conversions.py [hunk 2]
```diff
@@ -45,12 +45,24 @@ def generate_conversion_problem(start: Optional[str] = None, length: int = 3):
...
             end = prev
             continue
         length -= 1
 
-    return start,end
\ No newline at end of file
+    return start,end
+
+def generate_problem():
+	from random import choice
```
Comment (CodeWithSwastik) [inline conversions.py]: Could you add the documentation here?

File: conversions.py [hunk 3]
```diff
@@ -45,12 +45,24 @@ def generate_conversion_problem(start: Optional[str] = None, length: int = 3):
...
+	start = choice(list(reactions))
+	solution = choice(list(reactions[start]))
+	solutions = [solution]
+	for i in range(2):
+		buffer = choice(list(reactions[choice(list(reactions))]))
```
Comment (CodeWithSwastik) [inline conversions.py]: Wouldn't this raise an error if the reagent chosen wasn't in the reactions dict as a key?

**Comment by CodeWithSwastik** [comment]
I'll merge this when you make the changes and I'll change it when I get time

**Comment by sachin-sankar** [comment]
> I like this a lot but I'm planning to implement this in FastAPI and not Flask.

I will implement the same in FastAPI ASAP. And i will make the requested changes too.

File: conversions.py (hunk 1 above)
Comment (sachin-sankar) [reply conversions.py]: It's a formatting error will fix it.

**Comment by sachin-sankar** [comment]
@CodeWithSwastik Updated the files and made the requested changes.

**Pull request merged**